app/src/main/kotlin/com/wire/android/navigation/HomeDestination.kt

```diff
@@ -114,28 +127,40 @@ sealed class HomeDestination(
 
         fun values(): Array<HomeDestination> =
             arrayOf(Conversations, Favorites, Group, OneOnOne, Settings, Vault, Archive, Support, WhatsNew)
+
+        fun getArgumentsFromEntry(navBackStackEntry: NavBackStackEntry?): FolderNavArgs? {
+            return navBackStackEntry?.let {
+                if (it.destination.route == FolderConversationsScreenDestination.route) {
+                    FolderConversationsScreenDestination.argsFrom(it)
+                } else {
+                    null
+                }
+            }
+        }
```

Comment:
suggestion: the idea of taking args from entry is nice but it's a bit error-prone in case there are different types with the same arguments. I would suggest to merge that with the lambda that's using it in `HomeStateHolder` and make a function that already returns `HomeDestination` - WDYT?

I would do it by adding matching method and one open that returns the `HomeDestination` to the `HomeDestination`:
```
internal fun NavBackStackEntry.baseRouteMatches(): Boolean = 
        direction.route.getBaseRoute() == destination.route?.getBaseRoute()
open fun getFromEntry(entry: NavBackStackEntry): HomeDestination? =
        if (entry.baseRouteMatches()) this else null
``` 
so that all items by default make use of this method, but for `data class Folder` add custom implementation:
```
data class Folder(val folderNavArgs: FolderNavArgs) : HomeDestination(
        title = UIText.DynamicString(folderName),
        icon = R.drawable.ic_conversation,
        isSearchable = true,
        withNewConversationFab = true,
        direction = FolderConversationsScreenDestination(folderNavArgs)
    ) {
    override fun getFromEntry(entry: NavBackStackEntry): HomeDestination? =
            if (entry.baseRouteMatches()) Folder(FolderConversationsScreenDestination.argsFrom(entry)) else null
}
```